h5i_image.go

```diff
@@ -0,0 +1,79 @@
+// license that can be found in the LICENSE file.
+
+package hdf5
+
+// #include "hdf5.h"
+// #include "hdf5_hl.h"
+// #include <stdlib.h>
+// #include <string.h>
+import "C"
+
+import (
+	"errors"
+	"image"
+	"image/color"
+	"unsafe"
+)
+
+// newImage takes a image object and convert it to a hdf5 format and write to the id node
+func newImage(id C.hid_t, name string, img image.Image) error {
+	if img == nil {
+		return errors.New("nil image!")
```

Comment:
I think we can omit this error return and just panic at the next line. Please delete this conditional.